index.js

```diff
@@ -5,15 +5,35 @@ const through = require('through2');
 const debug = require('debug')('tcp-proxy');
 const EventEmitter = require('events').EventEmitter;
 
-function genThrough(interceptor) {
+function genThrough(interceptor, connection) {
+
   return through.obj(function(chunk, enc, done) {
-    const result = interceptor(chunk, enc);
+
+    const context = {
+      client: {
+        ip: connection.client.address().address,
+        port: connection.client.address().port,
+      },
+      server: {
+        ip: connection.server.address().address,
+        port: connection.server.address().port,
+      },
+      self: {
+        ip: connection.forwardHost,
+        port: connection.forwardPort,
+      },
+      size: chunk.length,
+    };
+
+    const result = interceptor(chunk, enc, context);
```

Comment:
You only need change to 

```js
const result = interceptor(chunk, enc, { client, server });
```

It's easier for extending. And because you can get `client` and `server` from the interceptor, so you can get `address` and `port` in your interceptor by `client.address()` and `server.address()`. Does it more reasonable?

Reply:
The syntax is heightened for no reason, on top of this the object is sent with information that isn't required, please read my above comment.